spec/unit/api_client/registration_spec.rb

```diff
@@ -94,6 +94,7 @@
 
   it "has an HTTP client configured with validator credentials" do
     expect(registration.http_api).to be_a_kind_of(Chef::ServerAPI)
+    expect(registration.http_api.options[:client_name]).to eq("test-validator")
```

Comment:
Might be worth getting the authenticator middleware and confirming that it has the client_name that you expect (cf. https://github.com/chef/chef/blob/9fd0dfdb1ce2e89c4e232f159d31a48b1ed014b1/lib/chef/http/authenticator.rb#L71-L73 ). It's kinda reaching into implementation detail, but since the API for middleware initialization is a free-form Hash, I'd prefer the extra assurance we're really configuring it correctly over unit test purity.